noir_stdlib/src/hash/poseidon/bn254/consts.nr

```diff
@@ -0,0 +1,179 @@
+// Constants for various Poseidon instances in the case of the prime field of the same order as BN254.
+// Consistent with https://github.com/iden3/circomlib/blob/master/circuits/poseidon.circom and https://github.com/iden3/circomlib/blob/master/circuits/poseidon_constants.circom
+
+// Number of full rounds
+global rf = 8;
+
+// Number of partial rounds
+fn rp() -> [u8; 16]
+{
+    [56, 57, 56, 60, 60, 63, 64, 63, 60, 66, 60, 65, 70, 60, 64, 68]
+}
+
+// S-box power
+global alpha = 5;
```

Comment:
alpha could be computed from the field (and so removed from bn254 by doing:
if -1 % 5 != 0 {
 5
} else {
constrain -1 %3  !=  0;
3
};
However it is probable that we can't have it as a global if we do this which will cause other issues

Reply:
Resolving as if are not compatible with global, and modulo does not work with field...